test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -813,6 +818,66 @@ UCS_TEST_P(test_uct_cm_sockaddr, conn_to_non_exist_ip)
     }
 }
 
+UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_with_delay)
+{
+    UCS_TEST_MESSAGE << "Testing "     << m_listen_addr
+                     << " Interface: " << GetParam()->dev_name;
+
+    m_delay_conn_reply = true;
+
+    cm_listen_and_connect();
+
+    EXPECT_FALSE(m_cm_state &
+                 (TEST_CM_STATE_SERVER_CONNECTED | TEST_CM_STATE_CLIENT_CONNECTED |
+                  TEST_CM_STATE_CLIENT_GOT_REJECT | TEST_CM_STATE_CLIENT_GOT_ERROR));
+
+    ASSERT_EQ(1ul, m_delayed_conn_reqs.size());
+
+    while (!m_delayed_conn_reqs.empty()) {
+        server_accept(m_server, m_delayed_conn_reqs.front(),
+                      server_connect_cb, server_disconnect_cb, this);
+        m_delayed_conn_reqs.pop();
+    }
+
+    wait_for_bits(&m_cm_state, TEST_CM_STATE_SERVER_CONNECTED |
+                               TEST_CM_STATE_CLIENT_CONNECTED);
+    EXPECT_TRUE(ucs_test_all_flags(m_cm_state, TEST_CM_STATE_SERVER_CONNECTED |
+                                               TEST_CM_STATE_CLIENT_CONNECTED));
+
+    cm_disconnect(m_client);
+}
+
+UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
+{
+    UCS_TEST_MESSAGE << "Testing "     << m_listen_addr
+                     << " Interface: " << GetParam()->dev_name;
+
+    ucs_status_t status;
+
+    m_delay_conn_reply = true;
+
+    cm_listen_and_connect();
+
+    EXPECT_FALSE(m_cm_state &
```

Comment:
seems like both tests could be united to single common helper function which takes an argument `bool reject`?